### ISSUE TITLE
[Bug]: System.MissingMethodException: Method not found. After upgrading to 7.1.2

### ISSUE THREAD

**Issue by RafaelLiendo** [reactiveui/refit#1756]
### Describe the bug 🐞

Hi, I've recently updated from `7.1.1` to `7.1.2` and started seeing the following runtime error:
```System.MissingMethodException: Method not found: 'Boolean Refit.IApiResponse`1.get_IsSuccessStatusCode()'.```

### Step to reproduce

My dependencies look like this:

## ProjectA.Api (net8.0)
- Refit.HttpClientFactory 7.1.1
  - Refit 7.1.1
- ProjectB.Client (net8.0)
  - Refit 7.1.0
---

And the error happens when I try to call a method in the `ProjectB.Client`

```csharp
namespace ProjectB.Client;

public interface IProjectBClient
{
    [Get("/" + ApiEndpoints.FetchSomethingById)]
    Task<IApiResponse<Something>> GetSomethingById(int somethingId, CancellationToken cancellationToken = default);
}
```

```csharp
IProjectBClient client; // from DI

var response = await client.GetSomethingById(123);
if(! tenantResponse.IsSuccessStatusCode) {
  throw new Exception("Failed to get Something");
}

### Reproduction repository

_No response_

### Expected behavior

Shouldn't have breaking changes when upgrading from version `7.1.1` to `7.1.2`

### Screenshots 🖼️

_No response_

### IDE

_No response_

### Operating system

_No response_

### Version

_No response_

### Device

_No response_

### Refit Version

7.1.2

### Additional information ℹ️

_No response_

**Comment by ChrisPulman** [comment]
Forgive me if I am wrong, do you have 1 system running 7.1.0 and another running 7.1.1, you then update to 7.1.2 and experience the issue. 
7.1.0 introduced a breaking change with IAPIResponce unintentionally which was then reversed with patch 7.1.2
If both systems are running with 7.1.2 do you still experience the same issue?

**Comment by RafaelLiendo** [comment]
> Forgive me if I am wrong, do you have 1 system running 7.1.0 and another running 7.1.1, you then update to 7.1.2 and experience the issue.

Correct, ProjectA and ProjectB are two microservices owned by different teams. ProjectB also publishes a client to nuget using Refit 7.1.0

I've created a repo to reproduce the issue:
https://github.com/RafaelLiendo/RefitIssue

> 7.1.0 introduced a breaking change with IAPIResponce unintentionally which was then reversed with patch 7.1.2

One idea to avoid this in the future would be to split the Refit package and have something like a `Refit.Annotations` package. This way, projects that only want to publish the models and the interface with annotations don't need to depend on the whole Refit package.

> If both systems are running with 7.1.2 do you still experience the same issue?

That solves the issue, but it's a bit hard to do it when the systems are owned by different teams. 
I also found other solutions and mentioned them in the repo above.

Thank you for you time🙏

**Comment by ChrisPulman** [comment]
We have API tracking now, so I hope this situation shouldn't occur again.
Apologies for any inconvenience caused.
Thank you for raising this issue, I will close it now.
7.1.2 is the correct package to use, 7.1.0 and 7.1.1 have been delisted.
